lib/manager/terraform/lockfile/util.ts

```diff
@@ -0,0 +1,204 @@
+import { readLocalFile } from '../../../util/fs';
+import { get as getVersioning } from '../../../versioning';
+import type { UpdateArtifactsResult } from '../../types';
+import type {
+  LineNumbers,
+  ProviderLock,
+  ProviderLockUpdate,
+  ProviderSlice,
+} from './types';
+
+const providerStartLineRegex = /^provider "(?<registryUrl>[^/]*)\/(?<namespace>[^/]*)\/(?<depName>[^/]*)"/;
+const versionLineRegex = /^(?<prefix>[\s]*version[\s]*=[\s]*")(?<version>[^"']+)(?<suffix>".*)$/;
+const constraintLineRegex = /^(?<prefix>[\s]*constraints[\s]*=[\s]*")(?<constraint>[^"']+)(?<suffix>".*)$/;
+const hashLineRegex = /^(?<prefix>\s*")(?<hash>[^"]+)(?<suffix>",.*)$/;
```

Comment:
@zharinov Can you create a `moo` lexer for this, so we don't need this complex regexes and parser functions?

I think  a `moo` parser would make this files a lot smaller and simplier.

Reply:
Sure

Reply:
Assign some issue to me maybe?

Reply:
There is no issue specifically, this is brand new and part of#7895